src/common/singleton.py

```diff
@@ -0,0 +1,7 @@
+class Singleton(type):
```

Comment:
I don't think this is thread safe singelton. We should do more research on singleton in python, but this should not be your task. We will have a look at it when we work together.

Reply:
We can start from [here](https://stackoverflow.com/questions/50566934/).